FirebaseDatabase/Tests/Unit/Swift/DatabaseAPITests.swift

```diff
@@ -0,0 +1,598 @@
+//
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// MARK: This file is used to evaluate the experience of using Firebase APIs in Swift.
+
+import Foundation
+
+import FirebaseCore
+import FirebaseDatabase
+
+final class DatabaseAPITests {
+  func usage() {
+    // MARK: - Database
+
+    // Retrieve Database Instance
+    _ = Database.database() as Database
+    _ = Database.database(url: "url" as String) as Database
+    if let app = FirebaseApp.app() {
+      _ = Database.database(app: app as FirebaseApp, url: "url" as String) as Database
+      _ = Database.database(app: app as FirebaseApp) as Database
+    }
+    let instance = Database.database() as Database
+    // Retrieve FirebaseApp
+    _ = instance.app as FirebaseApp?
+    // Retrieve DatabaseReference
+    _ = instance.reference() as DatabaseReference
+    _ = instance.reference(withPath: "path" as String) as DatabaseReference
+    _ = instance.reference(fromURL: "url" as String) as DatabaseReference
```

Comment:
@jrcrespoh So is this where a warning appears asking you to do the cast? Does doing `instance.reference(withPath: "path")` not work?

I would say keep it as is because, ideally, I think the API should be able to handle the literal. Showing the cast in the build test highlights some awkwardness in the API that we can improve.

Reply:
I don't need to cast for it to compile successfully, I chose to do so to be explicit about the type since some function calls used parameters of type `Any?` so I had to provide a literal and I arbitrarily used string literals, so I wanted to highlight the arbitrariness.

Also, the explicit type casting would provide compile-time errors whenever types used are changed (say from `Any` to `Any?`) so I thought that would be useful and thus went ahead and used type casting.

Reply:
Gotcha, I think an alternative approach that shows clearer usage while still checking explicit types could be:
```swift
let url = "url"
databaseRef = instance.reference(fromURL: url)
_ = databaseRef as DatabaseReference
```
WDYT?

Reply:
LGTM. How would you propose we tackle function calls that take in a parameter of type, say, `Any?`?